python/GafferSceneUI/SceneInspector.py

```diff
@@ -1182,20 +1209,22 @@ def update( self, targets ) :
 
 class _SectionWindow( GafferUI.Window ) :
 
-	def __init__( self, title, section, target ) :
+	def __init__( self, title, section, target, nodes = None ) :
```

Comment:
Did you add the `nodes` argument because you wanted to allow a diff view in this window? I don't think that will work when comparing two different paths from the same node - that requires multiple paths rather than multiple nodes. Specifying both `nodes` and `target` also makes it possible to provide nonsensical permutations, where the `nodes` don't correspond to `target.plug`. I think either we need to replace `target` with `targets` and drive the NodeSet from them, or we need to replace `target` with `targetPaths` and use that in conjunction with `nodes`. Unless there's a reason not to, I'd just go with `targets`.

Reply:
That's exactly why I added it. I think I misunderstood a bit how targets work and it's clearer now - thanks.

Changes in c7ad8f9. Note that later commits change some code in the same spot. I wasn't sure if it's easier to squash in all commits or go through it step by step.